survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/questions/DateTimePickerQuestionView.kt

```diff
@@ -0,0 +1,61 @@
+package com.quickbirdstudios.surveykit.backend.views.questions
+
+import android.content.Context
+import com.quickbirdstudios.surveykit.AnswerFormat
+import com.quickbirdstudios.surveykit.StepIdentifier
+import com.quickbirdstudios.surveykit.backend.views.question_parts.DateTimePickerPart
+import com.quickbirdstudios.surveykit.backend.views.step.QuestionView
+import com.quickbirdstudios.surveykit.result.QuestionResult
+import com.quickbirdstudios.surveykit.result.question_results.DateTimeQuestionResult
+
+internal class DateTimePickerQuestionView(
+    context: Context,
+    id: StepIdentifier,
+    isOptional: Boolean,
+    title: String?,
+    text: String?,
+    nextButtonText: String,
+    private val answerFormat: AnswerFormat.DateTimeAnswerFormat,
+    private val preselected: AnswerFormat.DateTimeAnswerFormat.DateTime?
+) : QuestionView(context, id, isOptional, title, text, nextButtonText) {
+
+    private lateinit var dateTimePickerPart: DateTimePickerPart
+
+    override fun createResults(): QuestionResult = DateTimeQuestionResult(
+        id = id,
+        startDate = startDate,
+        stringIdentifier = dateTimePickerPart.toDateTime().toString(),
+        answer = dateTimePickerPart.toDateTime()
+    )
+
+    override fun setupViews() {
+        super.setupViews()
+
+        dateTimePickerPart = content.add(DateTimePickerPart(context))
+        preselected?.let {
+            dateTimePickerPart.selectedTime = it.toTime()
+            dateTimePickerPart.selectedDate = it.toDate()
+        }
+        answerFormat.defaultValue?.let {
+            dateTimePickerPart.selectedTime = it.toTime()
+            dateTimePickerPart.selectedDate = it.toDate()
+        }
```

Comment:
Why did you reorder this?

If I select a datetime, go to the next step and go back again it always uses the datetime default and the previously selected one is lost.

Reply:
Hm I think I tried this to understand and I forgot to revert it back, sorry for forgetting to test this case

Reply:
There is nothing to be sorry about. I consider this a flaw of my architecture and I want to change it in the future so that making a mistake here is not possible anymore. 👍